packages/web-app-files/src/index.ts

```diff
@@ -33,60 +34,70 @@ const appInfo = {
   extensions: [],
   fileSideBars
 }
-const navItems = [
-  {
-    name(capabilities) {
-      return capabilities.spaces?.enabled ? $gettext('Personal') : $gettext('All files')
-    },
-    icon: appInfo.icon,
-    route: {
-      path: `/${appInfo.id}/spaces/personal`
-    }
-  },
-  {
-    name: $gettext('Favorites'),
-    icon: 'star',
-    route: {
-      path: `/${appInfo.id}/favorites`
+const navItems = (context) => {
+  return [
+    {
+      name(capabilities) {
+        return capabilities.spaces?.enabled ? $gettext('Personal') : $gettext('All files')
+      },
+      icon: appInfo.icon,
+      route: {
+        path: `/${appInfo.id}/spaces/personal`
+      },
+      enabled(capabilities) {
+        if (!capabilities.spaces?.enabled) {
+          return true
+        }
+        return !!context?.$store?.getters['runtime/spaces/spaces'].find((drive) =>
+          isPersonalSpaceResource(drive)
+        )
+      }
     },
-    enabled(capabilities) {
-      return capabilities.files && capabilities.files.favorites
-    }
-  },
-  {
-    name: $gettext('Shares'),
-    icon: 'share-forward',
-    route: {
-      path: `/${appInfo.id}/shares`
+    {
+      name: $gettext('Favorites'),
+      icon: 'star',
+      route: {
+        path: `/${appInfo.id}/favorites`
+      },
+      enabled(capabilities) {
+        return capabilities.files && capabilities.files.favorites
+      }
     },
-    activeFor: [{ path: `/${appInfo.id}/spaces/share` }],
-    enabled(capabilities) {
-      return capabilities.files_sharing?.api_enabled !== false
-    }
-  },
-  {
-    name: $gettext('Spaces'),
-    icon: 'layout-grid',
-    route: {
-      path: `/${appInfo.id}/spaces/projects`
+    {
+      name: $gettext('Shares'),
+      icon: 'share-forward',
+      route: {
+        path: `/${appInfo.id}/shares`
+      },
+      activeFor: [{ path: `/${appInfo.id}/spaces/share` }],
+      enabled(capabilities) {
+        return capabilities.files_sharing?.api_enabled !== false
+      }
     },
-    activeFor: [{ path: `/${appInfo.id}/spaces/project` }],
-    enabled(capabilities) {
-      return capabilities.spaces && capabilities.spaces.projects === true
-    }
-  },
-  {
-    name: $gettext('Deleted files'),
-    icon: 'delete-bin-5',
-    route: {
-      path: `/${appInfo.id}/trash/overview`
+    {
+      name: $gettext('Spaces'),
+      icon: 'layout-grid',
+      route: {
+        path: `/${appInfo.id}/spaces/projects`
+      },
+      activeFor: [{ path: `/${appInfo.id}/spaces/project` }],
+      enabled(capabilities) {
+        return capabilities.spaces && capabilities.spaces.projects === true
```

Comment:
```suggestion
        return capabilities.spaces?.projects
```
while you're already at it :-)